frontends/web/src/routes/account/add/add.tsx

```diff
@@ -109,29 +100,71 @@ class ManageAccount extends Component<Props, State> {
                 apiPost('account-add', {
                     coinCode,
                     name: accountName,
-                }).then((data: ResponseData) => {
-                    if (data.success) {
-                        this.setState({ step: 2 });
-                        //route('/account/' + data.accountCode);
-                    } else {
-                        if (data.errorCode) {
-                            alertUser(this.props.t(`addAccount.error.${data.errorCode}`));
-                        } else if (data.errorMessage) {
-                            alertUser(this.props.t('unknownError', { errorMessage: data.errorMessage }));
+                })
+                    .then((data: ResponseData) => {
+                        if (data.success) {
+                            this.setState({ step: 'success' });
+                            //route('/account/' + data.accountCode);
+                        } else {
+                            this.showError(data);
                         }
-                    }
-                });
+                    })
+                    .catch(error => this.showError(error));
                 break;
             case 'success':
+                // TODO: route to manage accounts to configure tokens
                 route('/account-summary');
                 return;
         }
     }
 
+    private showError = (error) => {
+        const { t } = this.props;
+        if (!error) {
+            return;
+        }
+        if (error.errorCode) {
+            this.setState({
+                errorMessage: t(`addAccount.error.${error.errorCode}`)
+            });
+        } else if (error.errorMessage) {
+            this.setState({
+                errorMessage: t('unknownError', { errorMessage: error.errorMessage })
+            });
+        } else if (error) {
+            this.setState({
+                errorMessage: error
+            });
+        }
+    }
+
+    private isFirstStep = () => {
+        switch (this.state.step) {
+            case 'select-coin':
+                return true;
+            case 'choose-name':
+                return this.onlyOneSupportedCoin();
+            case 'success':
+                return false;
+        }
+    }
+
+    private focusRef = (ref) => {
```

Comment:
can this be moved into the Input component? we can use it in lots of other places. e.g. `<Input autoFocus ...>`.

Reply:
maybe, but I don't want to loose too much time with focus with preact 8 anymore.

Reply:
Maybe it already works, like this?

https://github.com/digitalbitbox/bitbox-wallet-app/blob/staging-multiaccounts/frontends/web/src/components/backups/create.jsx#L91-L92

Reply:
This doesn't work afaik if another element  already has focus.